### ISSUE TITLE
BUG: add testEnvironment: 'node' to jest.config.js

### ISSUE THREAD

**Issue by lightningRalf** [lightningRalf/obsidian-excalidraw-plugin#4]
For testing the backend with the file connectObjects.test.ts - please adjust the jest.config.js:
```
module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  moduleNameMapper: {
    '^@/(.*)$': '<rootDir>/src/$1',
  },
};
```

Also add a 'decider' when to youse what module.exports

**Comment by lightningRalf** [comment]
To handle different test environments for different tests in Jest, you can configure the testEnvironment directly in your test files or use a jest.config.js file to specify different environments based on the location of your test files.

Here's how you can achieve this:

Using jest.config.js to set environments based on folder structure:

In your jest.config.js, you can use a projects configuration to specify different settings for different sets of tests.

module.exports = {
  projects: [
    {
      displayName: 'jsdom',
      testEnvironment: 'jsdom',
      testMatch: ['<rootDir>/__tests__/jsdom/**/*.test.js']
    },
    {
      displayName: 'node',
      testEnvironment: 'node',
      testMatch: ['<rootDir>/__tests__/node/**/*.test.js']
    }
  ]
};

This way, tests located in the __tests__/jsdom folder will run in the jsdom environment, and tests in the __tests__/node folder will run in the node environment.

Setting testEnvironment directly in individual test files:

If you prefer, you can set the testEnvironment at the top of each test file using a comment:

// __tests__/someTest.test.js
/**
 * @jest-environment jsdom
 */

// Your test code here

// __tests__/anotherTest.test.js
/**
 * @jest-environment node
 */

// Your test code here

Using package.json:

You generally don't need to specify environments in package.json. However, if you want to use custom scripts, you can add scripts that point to different configurations:

{
  "scripts": {
    "test:jsdom": "jest --config=jest.config.js --testPathPattern=__tests__/jsdom",
    "test:node": "jest --config=jest.config.js --testPathPattern=__tests__/node"
  }
}

To summarize, the recommended approach is to use a jest.config.js file with a projects configuration to handle multiple environments based on the test file location. This keeps your setup clean and easy to manage.
